### ISSUE TITLE
Manage user sessions [question] [help wanted]

### ISSUE THREAD

**Issue by TheKnown44** [voxpelli/node-connect-pg-simple#195]
Hi, 

I'm developing a website with node.js, passport.js, express and express-sessions. 

I want to make a feature that allows signed-in users to remove previous sessions from e.g. other devices they signed into before and forgot to logout from, basically so they could choose to logout from all devices or specific devices. I also want to store user-agent and IP address in the database along with the user sessions.

Express-sessions are stored in a database table using the [connect-pg-simple][2] store.

The session table has 3 columns and looks like this:

|  sid | sess | expire |
|:---- |:------:| -----:|
| AH_oe3YCEahqcKA2VxxHp_Xf  | {"cookie":{"originalMaxAge":7200000,"expires":"2021-01-15T11:11:02.164Z","secure":false,"httpOnly":true,"path":"/"},"flash":{},"passport":{"user":"106"}}| 2021-01-15 12:11:03 |
     

Currently the userid is stored inside a JSON string `{"user":"106"}` on the session, so I don't know how to query all sessions that belong to a specific user. I would make a new column `userId` along with the 3 existing columns and store userId, but I don't know if that's possible at all and I don't know how I would be able to save user-agent and IP address at the same time.

I will gladly provide more code that I have or information about the packages that I have used. I have searched everywhere, but I couldn't find any helpful information.

Any help would be appreciated. Thanks.

  [2]: https://www.npmjs.com/package/connect-pg-simple

**Comment by iwaduarte** [comment]
So you could use set and get methods to alter the session the way you want.
Remember that is just a table on the database with parameters with JSON stringified. You can do whatever you want.

**Comment by voxpelli** [comment]
Yeah, set the data on the session and then if you want to have it saved in individual columns, then currently you would have to extend the class and override the `set` method within.

I guess we could add a callback for deriving such additional values to add to individual columns, would be a fairly simple little addition.

**Comment by TheKnown44** [comment]
Thank you. I managed to solve it for now.

**Comment by TheKnown44** [comment]
Hi @voxpelli,

I have run into issues and my solution didn't work that well.

Currently I do it like this `SELECT FROM sessions WHERE sess->'passport'->>'user' = $1`

Could you please explain how to extend the class and override the set method? (By class you mean the PGStore class?)

All I want is to have userid in a separate column so I can more easily query the sessions for specific users. 

Thanks.
